solanaj/src/main/java/org/p2p/solanaj/kits/transaction/parser/ConfirmedTransactionRootParser.kt

```diff
@@ -0,0 +1,201 @@
+                )
+            }
+            "closeAccount" -> {
+                parseCloseTransaction(
+                    parsedInstruction = parsedInstruction,
+                    parsedInfo = parsedInfo,
+                    transactionRoot = transactionRoot,
+                    signature = signature,
+                )
+            }
+            "create" -> {
+                parseCreateAccountTransaction(
+                    signature = signature,
+                    transactionRoot = transactionRoot
+                )
+            }
+            else -> {
+                parseUnknownTransaction(
+                    signature = signature,
+                    transactionRoot = transactionRoot
+                )
+            }
+        }
+    }
+
+    private fun parseCreateAccountTransaction(
+        signature: String,
+        transactionRoot: ConfirmedTransactionRootResponse
+    ): CreateAccountDetails {
+        return CreateAccountDetails(
+            signature = signature,
+            slot = transactionRoot.slot,
+            blockTime = transactionRoot.blockTime,
+            fee = transactionRoot.meta.fee
+        )
+    }
+
+    private fun parseUnknownTransaction(
+        signature: String,
+        transactionRoot: ConfirmedTransactionRootResponse
+    ): UnknownDetails {
+        return UnknownDetails(
+            signature = signature,
+            blockTime = transactionRoot.blockTime,
+            slot = transactionRoot.slot
+        )
+    }
+
+    private fun parseCloseTransaction(
+        parsedInstruction: InstructionResponse,
+        parsedInfo: InstructionInfoDetailsResponse,
+        transactionRoot: ConfirmedTransactionRootResponse,
+        signature: String,
+    ): CloseAccountDetails? {
+        if (parsedInstruction.programId != SPL_TOKEN_PROGRAM_ID.toBase58()) {
+            return null
+        }
+
+        val closedTokenPublicKey = parsedInfo.info.account
+        val preBalances = transactionRoot.meta.preTokenBalances.firstOrNull()?.mint
+        return CloseAccountDetails(
+            signature = signature,
+            blockTime = transactionRoot.blockTime,
+            slot = transactionRoot.slot,
+            account = closedTokenPublicKey,
+            mint = preBalances
+        )
+    }
+
+    private fun parseTransferTransaction(
+        signature: String,
+        transactionRoot: ConfirmedTransactionRootResponse,
+        parsedInfo: InstructionInfoDetailsResponse,
+    ): TransferDetails {
+        val info = parsedInfo.info
+
+        val mint: String?
+        val amount: String?
+        val decimals: Int
+        if (parsedInfo.type == "transferChecked") {
+            mint = info.mint
+            amount = info.tokenAmount?.amount
+            decimals = info.tokenAmount?.decimals?.toInt() ?: 0
+        } else {
+            mint = null
+            amount = info.lamports?.let { BigInteger.valueOf(it.toLong()).toString() } ?: info.amount
```

Comment:
info.lamports?.toBigInteger() ? or it is not working?

Reply:
are you sure that we have this extension in SOLANAJ module?